src/client/Client.cc

```diff
@@ -9858,20 +9858,20 @@ void Client::_release_filelocks(Fh *fh)
   list<pair<int, ceph_filelock> > to_release;
 
   if (fh->fcntl_locks) {
-    ceph_lock_state_t* lock_state = fh->fcntl_locks;
+    auto &lock_state = fh->fcntl_locks;
     for(multimap<uint64_t, ceph_filelock>::iterator p = lock_state->held_locks.begin();
 	p != lock_state->held_locks.end();
 	++p)
       to_release.push_back(pair<int, ceph_filelock>(CEPH_LOCK_FCNTL, p->second));
-    delete fh->fcntl_locks;
+    lock_state.reset();
```

Comment:
Is this equivalent to `lock_state.reset(nullptr);` ? I hope so :-)

http://en.cppreference.com/w/cpp/memory/unique_ptr/reset

Reply:
Yes, it is.